### PR TITLE
extract the plist via ASN1 parser

### PR DESCRIPTION
**Pull request by tcurdt**
## Change description

Parse the provisioning profile (properly) to extract the plist.
Added an ASN1 parser as dependency.

## Test Plan

The same tests are still passing.

## Type of Change

- [ ] Bug fix
- [ ] Feature
- [x] Enhancement 

## Guidelines

- [ ] A [changeset] is included, or the change is not noteworthy enough to warrant one

[changeset]: https://github.com/changesets/changesets/blob/main/docs/adding-a-changeset.md#i-am-in-a-single-package-repository

### CODE REVIEW

File: Source/EmbeddedProvision.swift [hunk 1]
```diff
@@ -1,5 +1,8 @@
+import ASN1Decoder
 import Foundation
 
+// import Security
+
```
Comment (stigi) [inline Source/EmbeddedProvision.swift:5]: looks like some leftover

File: Package.swift [hunk 2]
```diff
@@ -7,22 +7,30 @@ let package = Package(
...
         .iOS(.v12),
-        .macOS(.v10_15)
+        .macOS(.v10_15),
     ],
     products: [
         .library(
             name: "EmbeddedProvision",
             targets: ["EmbeddedProvision"]
-        ),
+        )
```
Comment (stigi) [inline Package.swift:16]: Thanks for cleaning up that comma, but why did you move it to line 10? ;)

**Review by stigi — APPROVED** [review]
Awesome! Works as expected and is much nicer this way! 👏 

Could you add a changeset via `yarn && yarn changeset` please? That way the change will be attibuted to you. I think a patch level change should be enough, as no API changes occured.

File: Source/EmbeddedProvision.swift (hunk 1 above)
Comment (tcurdt) [reply Source/EmbeddedProvision.swift:5]: Yes, I have some code to check the code signing certificate to see if it's a testflight build - but I left that out for now.

File: Package.swift (hunk 2 above)
Comment (tcurdt) [reply Package.swift:16]: I didn't do anything - my editor auto formatted :)
Do you have some formatting settings to apply?

**Comment by tcurdt** [comment]
> Could you add a changeset via `yarn && yarn changeset` please?

I can try. Never used that before.

Would be good to give some more instructions in the `contribution.md`.
Even this https://github.com/changesets/changesets/blob/main/docs/adding-a-changeset.md does not feel good enough.

File: Package.swift (hunk 2 above)
Comment (stigi) [reply Package.swift:16]: Indeed, running `swift format -i --recursive Sources Tests Package.swift` applies all that. I'll automate formatting in a pre-commit hook in a seperate PR, and will reformat everything (lot's of changes in test files).

It's the same formatting that's built in in Xcode 16 via `ctrl+shift+i`.

**Comment by stigi** [comment]
@tcurdt I'll make sure to provide more on how to use changesets. I'll add one for this PR myself and will tag you as a contributor. Thank's for your time amnd effort. Don't want to keep you busy any longer and will merge the PR now. 🙌
